TWLight/users/views.py

```diff
@@ -532,4 +539,35 @@ def get_success_url(self):
                   'the terms of use.')
 
             messages.add_message(self.request, messages.WARNING, fail_msg)
-            return reverse_lazy('users:home')
\ No newline at end of file
+            return reverse_lazy('users:home')
+
+
+class AuthorizedUsers(APIView):
+    """
+    API endpoint returning the list of users authorized to access a specific
+    partner. For proxy partners, uses the list of Authorizations. For others,
+    uses the full list of sent applications.
+    """
+    authentication_classes = (TokenAuthentication,)
```

Comment:
Not super familiar with this code style for auth setup. My recollection is using decorators and mixins to achieve auth bliss. This isn't a change request, but you might just double check and make sure that we don't accidentally create a loophole where it's easy for users to get tokens generated and view stuff they shouldn't.

I'd usually expect some kind of permission table to be tied to the token to make sure that we scope what data the user can fetch, but we literally only have one thing right now. It might be worth sticking that in a TODO comment somewhere.

Reply:
This is how it's handled for class-based views, I think decorators is how it's done for function views. The docs don't suggest any mixins for authentication purposes (https://www.django-rest-framework.org/api-guide/authentication/), though it's possible to set authentication globally rather than for the individual view.